x-pack/plugins/ml/public/components/rule_editor/rule_editor_flyout.js

```diff
@@ -0,0 +1,526 @@
+
+            <EuiTitle>
+              <h2>Action</h2>
+            </EuiTitle>
+            <ActionsSection
+              actions={rule.actions}
+              onSkipResultChange={this.onSkipResultChange}
+              onSkipModelUpdateChange={this.onSkipModelUpdateChange}
+            />
+
+            <EuiSpacer size="xl" />
+
+            <EuiTitle>
+              <h2>Conditions</h2>
+            </EuiTitle>
+            <EuiSpacer size="s" />
+            <EuiCheckbox
+              id="enable_conditions_checkbox"
+              className="scope-enable-checkbox"
+              label="Add numeric conditions to take action according to the actual or typical values of the anomaly"
+              checked={isConditionsEnabled}
+              onChange={this.onConditionsEnabledChange}
+              disabled={!hasPartitioningFields}
+            />
+            <EuiSpacer size="s" />
+            <ConditionsSection
+              isEnabled={isConditionsEnabled}
+              conditions={rule.conditions}
+              addCondition={this.addCondition}
+              updateCondition={this.updateCondition}
+              deleteCondition={this.deleteCondition}
+            />
+
+            <EuiSpacer size="xl" />
+
+            <ScopeSection
+              isEnabled={isScopeEnabled}
+              onEnabledChange={this.onScopeEnabledChange}
+              partitioningFieldNames={this.partitioningFieldNames}
+              filterListIds={filterListIds}
+              scope={rule.scope}
+              updateScope={this.updateScope}
+            />
+
+            <EuiCallOut
+              title="Rerun job"
+              color="warning"
+              iconType="help"
+            >
+              <p>
+                Changes to rules take effect for new results only.
+              </p>
+              <p>
+                To apply these changes to existing results you must clone and rerun the job.
+                Note rerunning the job may take some time and should only be done once
+                you have completed all your changes to the rules for this job.
+              </p>
+            </EuiCallOut>
+
+          </EuiFlyoutBody>
+
+          <EuiFlyoutFooter>
+            <EuiFlexGroup justifyContent="spaceBetween">
+              <EuiFlexItem grow={false}>
+                <EuiButtonEmpty
+                  iconType="cross"
+                  onClick={this.closeFlyout}
+                  flush="left"
+                >
+                  Close
+                </EuiButtonEmpty>
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiButton
+                  onClick={this.saveEdit}
+                  isDisabled={!isValidRule(rule)}
+                  fill
+                >
+                  Save
+                </EuiButton>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiFlyoutFooter>
+        </EuiFlyout>
+      );
+
+    }
+
+    return (
+      <div>
```

Comment:
The PR uses a mix of `<div>` and `<React.Fragment>` in different places to wrap components. If the `<div>` isn't necessary in the DOM it could be replaced with `<React.Fragment>`.